plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/github.ts

```diff
@@ -263,7 +263,22 @@ export function createPublishGithubAction(options: {
               allow_rebase_merge: allowRebaseMerge,
             });
 
-      const { data: newRepo } = await repoCreationPromise;
+      let newRepo;
+
+      try {
+        newRepo = (await repoCreationPromise).data;
+      } catch (e) {
+        assertError(e);
+        if (e.message === 'Resource not accessible by integration') {
+          ctx.logger.warn(
+            `The GitHub app or token provided may not have the required permissions to create the ${user.data.type} repository ${owner}/${repo}.`,
+          );
+        }
+        throw new Error(
+          `Failed to create the ${user.data.type} repository ${owner}/${repo}: ${e.message}`,
```

Comment:
Ah, realized a tiny nit which is that we try to avoid separating error message sentences with `:`, since we already separate the error name from the message with a `:`. Thinking just a `,` makes sense in this case. Otherwise you can also use `ForwardedError` from @backstage/errors, but that gets a bit more verbose and seeing as this in the end is for the users I think it might be best to avoid.

Reply:
done thanks.